docs/types/value-types.rst

```diff
@@ -349,13 +349,9 @@ Contract Types
 
 Every :ref:`contract<contracts>` defines its own type.
 You can implicitly convert contracts to contracts they inherit from.
-Contracts can be explicitly converted to and from the ``address`` type.
-
-Explicit conversion to and from the ``address payable`` type is only possible
-if the contract type has a receive or payable fallback function.  The conversion is still
-performed using ``address(x)``. If the contract type does not have a receive or payable
-fallback function, the conversion to ``address payable`` can be done using
-``payable(address(x))``.
+Contracts can be explicitly converted to and from the ``address`` type using ``address(x)``.
+Explicit conversion to and from the ``address payable`` can be done with ``payable(address(x))``, regardless 
```

Comment:
Whilst this is technically true, from Solidity 0.8.0 and up, if you want to convert a contract into `address payable`, the `payable(x)` approach should be used as it's the safest way, i.e. it will only perform the conversion if the contract to be converted has a receive/fallback function implemented, otherwise, it will throw an error.

Using `payable(address(x))` does not offer such safety, although it does allow any contract to be converted into `address payable`, and should therefore be avoided.

Reply:
```solidity
contract Test {
    uint public test;

    receive() external payable {}

    fallback() external payable {}
}

contract Test2 {
    address payable a = payable(Test);
}
```
Could you elaborate? I can't seem to successfully do such a conversion with just payable(x) as you mentioned. The contract above isn't compiling with solidity 0.8.16.

Reply:
The `Test` you're trying to convert via `address payable a = payable(Test);` is a type, not an instance of said type.

Try 
```solidity
contract Test2 {
    Test test = new Test();
    address payable a = payable(test);
}
```

Reply:
I see, it works. Sorry for the confusion. But I also see that the documentation says the conversion is performed using address(x), and not payable(x)?

Reply:
Yea that's for the unsafe conversion. Is there nothing about the safe conversion that @nikola-matic was talking about?